src/Goteo/Controller/ChannelController.php

```diff
@@ -134,6 +137,23 @@ public function indexAction($id)
             $list = Project::published(['type' => 'random'], $id, 0, $limit);
         }
 
+        $values = $channel->getSections('values');
+        if ($values) {
+            $footprints = Footprint::getList();
+            $projects_by_footprint = [];
+            $sdg_by_footprint = [];
+            $footprintImpactData = [];
+            foreach($footprints as $footprint) {
+                foreach ($footprint->getAllImpactData() as $impactData) {
+                    if ($channel->hasImpactData($impactData)) {
+                        $footprintImpactData[$footprint->id][] = $impactData;
+                    }
+                }
+                $projects_by_footprint[$footprint->id] = Project::getByFootprintOrSDGs(['footprints' => $footprint->id, 'channel' => $channel->id]);
+                $sdg_by_footprint[$footprint->id] = Sdg::getList(['footprint' => $footprint->id]);
+            }
+        }
```

Comment:
It'll be nicer to have this inside a method. Perhaps, even in the `Project` and `Sdg` model. This way we keep the controller concise and brief.

There could be a method that obtains all Projects by `Footprint` in `Project` model. And another method that returns the `Sdg` by Footprint in the `Sdg` model.